### PR TITLE
[NT-647] "Thanks Page Viewed" event

### PR DESCRIPTION
**Pull request by ifbarrera**
# 📲 What

Adds the `Thanks Page Viewed` event.

# 🤔 Why

Tracking events cleanup.

# 🛠 How

Adds the `Thanks Page Viewed` event which includes `projectProperties`, `pledgeProperties` and `checkoutProperties`. Because the checkout properties are mostly defined in `PledgeViewModel`, they're passed through to the `ThanksViewModel` via the configuration function.

# ✅ Acceptance criteria

With the `KOALA_ENABLED` environment variable set to `true`, test the event by backing any project. You should see the `Thanks Page Viewed` event fire when you get to the thanks page, with the following properties:

- [x] `checkout_amount` // pledge total including shipping
- [x] `checkout_payment_type` // either `CREDIT_CARD` or `APPLE_PAY`
- [x] `checkout_reward_id`
- [x] `checkout_reward_title`
- [x] `checkout_shipping_amount`
- [x] `checkout_revenue_in_usd_cents`
- [x] `checkout_reward_estimated_delivery_on`
- [x] `checkout_reward_shipping_enabled`
- [x] `checkout_user_has_eligible_stored_apple_pay_card`

as well as all `projectProperties` and `pledgeProperties`.

Note that you'll only see `checkoutProperties` when backing a project through the native pledge view experience, although the `Thanks Page Viewed` event will also fire through the deprecated checkout experience.

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Library/ViewModels/PledgeViewModelTests.swift [hunk 1]
```diff
@@ -1185,15 +1204,15 @@ final class PledgeViewModelTests: TestCase {
...
 
       self.vm.inputs.paymentAuthorizationViewControllerDidFinish()
 
-      self.goToThanks.assertDidNotEmitValue()
+      self.goToThanksProject.assertDidNotEmitValue()
 
       self.scheduler.run()
 
-      self.goToThanks.assertValues([project])
+      self.goToThanksProject.assertValues([project])
```
Comment (Scollaco) [inline Library/ViewModels/PledgeViewModelTests.swift]: Should we assert the values of `goToThanksReward` and `goToThanksCheckoutData` here too? 🤔

File: Library/ViewModels/PledgeViewModelTests.swift [hunk 2]
```diff
@@ -1317,15 +1336,15 @@ final class PledgeViewModelTests: TestCase {
...
+      self.goToThanksProject.assertDidNotEmitValue()
       self.showErrorBannerWithMessage.assertDidNotEmitValue()
 
       self.scheduler.run()
 
       self.beginSCAFlowWithClientSecret.assertDidNotEmitValue()
       self.submitButtonEnabled.assertValues([false, true, false, true])
       self.submitButtonIsLoading.assertValues([true, false])
-      self.goToThanks.assertValues([.template])
+      self.goToThanksProject.assertValues([.template])
```
Comment (Scollaco) [inline Library/ViewModels/PledgeViewModelTests.swift]: And here too

File: Library/ViewModels/PledgeViewModelTests.swift [hunk 3]
```diff
@@ -3095,7 +3114,7 @@ final class PledgeViewModelTests: TestCase {
 
       self.beginSCAFlowWithClientSecret.assertValues(["client-secret"])
       self.submitButtonEnabled.assertValues([false, true, false, true])
-      self.goToThanks.assertValues([project])
+      self.goToThanksProject.assertValues([project])
```
Comment (Scollaco) [inline Library/ViewModels/PledgeViewModelTests.swift]: and here

File: Library/ViewModels/DeprecatedCheckoutViewModelTests.swift [hunk 4]
```diff
@@ -15,7 +15,9 @@ final class DeprecatedCheckoutViewModelTests: TestCase {
   fileprivate let evaluateJavascript = TestObserver<String, Never>()
   fileprivate let goToPaymentAuthorization = TestObserver<NSDictionary, Never>()
   fileprivate let goToSafariBrowser = TestObserver<URL, Never>()
-  fileprivate let goToThanks = TestObserver<Project, Never>()
+  fileprivate let goToThanksProject = TestObserver<Project, Never>()
+  fileprivate let goToThanksReward = TestObserver<Reward, Never>()
+  fileprivate let goToThanksCheckoutData = TestObserver<Koala.CheckoutPropertiesData?, Never>()
```
Comment (Scollaco) [inline Library/ViewModels/DeprecatedCheckoutViewModelTests.swift]: Could you alpha here? 🙈

File: Library/ViewModels/DeprecatedRewardPledgeViewModelTests.swift [hunk 5]
```diff
@@ -53,7 +53,9 @@ internal final class DeprecatedRewardPledgeViewModelTests: TestCase {
   fileprivate let goToShippingPickerProject = TestObserver<Project, Never>()
   fileprivate let goToShippingPickerShippingRules = TestObserver<[ShippingRule], Never>()
   fileprivate let goToShippingPickerSelectedShippingRule = TestObserver<ShippingRule, Never>()
-  fileprivate let goToThanks = TestObserver<Project, Never>()
+  fileprivate let goToThanksProject = TestObserver<Project, Never>()
+  fileprivate let goToThanksReward = TestObserver<Reward, Never>()
+  fileprivate let goToThanksCheckoutData = TestObserver<Koala.CheckoutPropertiesData?, Never>()
```
Comment (Scollaco) [inline Library/ViewModels/DeprecatedRewardPledgeViewModelTests.swift]: Alpha here too

File: Library/ViewModels/PledgeViewModelTests.swift [hunk 6]
```diff
@@ -43,7 +43,9 @@ final class PledgeViewModelTests: TestCase {
   private let goToApplePayPaymentAuthorizationShippingRule = TestObserver<ShippingRule?, Never>()
   private let goToApplePayPaymentAuthorizationMerchantId = TestObserver<String, Never>()
 
-  private let goToThanks = TestObserver<Project, Never>()
+  private let goToThanksProject = TestObserver<Project, Never>()
+  private let goToThanksReward = TestObserver<Reward, Never>()
+  private let goToThanksCheckoutData = TestObserver<Koala.CheckoutPropertiesData?, Never>()
```
Comment (Scollaco) [inline Library/ViewModels/PledgeViewModelTests.swift]: And here

**Pull request merged**